### ISSUE TITLE
on_schedule not supported in python?

### ISSUE THREAD

**Issue by maxtheman** [cloudflare/python-workers-examples#13]
Hi Team CF.

I don't think that cron triggers are supported in python workers today.

I ran this to test:

```
npx wrangler dev --test-scheduled

curl "http://localhost:8787/__scheduled?cron=*+*+*+*+*"
```

and used this as the handler:
```
async def on_scheduled(event, env: Env, context):
    print(f"Scheduled: {event}")
    return Response.json({"message": "Scheduled"})
```

And the response I got was this:

```
curl "http://localhost:8787/__scheduled?cron=*+*+*+*+*"
TypeError: Cannot read properties of undefined (reading 'callRelaxed')
    at async Object.fetch
```

Which is the same as if I have no code in `entry.py` at all.

I also tried adding this to wrangler.toml

```
[env.dev.triggers]
crons = ["0 * * * *"]
```
But that doesn't make any difference to the output on startup or upon sending the GET request.

It's possible I'm doing something wrong, if that's the case I'd appreciate the correction.

Let me know if there's a better repo for this.

**Comment by dom96** [comment]
Apologies it took so long for us to look at this. It does seem like something is wrong here, though it may be just this `--test-scheduled` feature that is broken. Can you try deploying your worker and see if your cron works?

**Comment by maxtheman** [comment]
Hey @dom96 , really appreciate the reply. I've moved on from this project, actually.

Unfortunately I've concluded that the beta python workers is just unusable right now, at least for R2 purposes.

Just to share my experience:
- I kept running into 500 errors that didn't log in the cloudflare-provided server logs and didn't reproduce locally. As such, troubleshooting was impossible.
- The only thing that I am able to guess, after ruling out everything else, was that I was hitting OOM errors with the already-large python memory footprint + video file handling.
- I subscribed to a business member plan for support on this, and other issues, but the support portal was broken (integration between Cloudflare and SFDC chat was messed up, possibly still is) and nobody ever replied to my cases.
- I charged back the business membership through mercury and won that case.

Hope this feedback can get back to the team @dom96, I tried to share in Discord, via email and in the community forums, but like I said -- nobody from cloudflare ever replied.

Also writing to document that I would strongly discourage anyone who comes across this thread from using python on cloudflare workers.

**Comment by maxtheman** [comment]
Thanks @dom96 I know it's beta software and this comes with the territory, so I wanted to share. The support experience was much worse than any failings of the python instance, which aside from this error was actually fairly good and technically impressive.

I should have mentioned — I did open source the project: https://github.com/maxtheman/r2_worker_server_python this is the exact code I was running. It should reproduce simply by getting that running and sending some files at it..

The original service is still running on workers. If you are interested I can share an api key via DM or Discord to interact with it directly so you can see the strange 502. Or, it should be straightforward to get running from the github project -- but it will not trigger the error running locally on wrangler, only when deployed.

Here's one of my forum posts about it, containing an exact description of the error.
 - https://community.cloudflare.com/t/error-1104-500-error-that-doesnt-show-up-in-worker-logs/695867
 - Quite similar issue: https://community.cloudflare.com/t/r2-intermittent-500-errors-on-put/485134/7

Thanks for the engagement and if you do investigate, happy to assist.

**Comment by maxtheman** [comment]
Hey @mikenomitch thank you for the continued follow up. This was a few months ago now, and I do tend to live on the bleeding edge of stuff so I am not so worried about the python bugs, you'll figure it out.

I was pretty convinced it was an out of memory issue because that is the only thing that I think would differ between the emulator running on my environment versus the constraints of the actual cloudflare system, and the error messages you get using Python are quite different from the ones that surface during a typical Cloudflare worker deployment, often times, due to that translation layer, so It would not at all shock me if some "normal" out of memory error was simply not surfaced in the correct way to the dev. Could be wrong about that.

Regarding support issues:

Yes I opened a few tickets and, even upgraded to business to do so. Unfortunately chat was not accessible and replies were unacceptably slow or non-existent via email.
- First ticket was: ID 01066137 - I did eventually get one response confirming to me that there's some issue, 24 days after submitting the initial response. I had already cancelled my business plan at that point and simply asked for help getting a refund, which they never replied to.
- Another: 01063295 - follow up to the original trying to get a response filed about a day later. Took ~15 days to get a reply. I didn't reply to this one.
- Discord messages: https://discord.com/channels/595317990191398933/1281410280877985853/1281410580057554966 the folks in discord were quite helpful and told me they escalated it, but didn't help to accelerate a reply from support.
- Final ticket - 01067348 me asking for a refund. I assumed I wouldn't get a response, but naturally got one within 24 hours denying the refund. I disputed the transaction with my bank already and got that settled in my favor, so not asking you to look into this per-say — just sharing because clearly _some_ of my tickets were going through.

I probably could have kept submitting tickets, or I do know some people who work at cloudflare IRL who I could have messaged, but honestly it just inspired zero confidence so I just wanted to take the project to a vendor I knew could support it properly if issues came up.

No slight to you or the python team at all intended, overall I was pretty impressed with the initial deployment experience and if I hadn't hit this specific error, I never would have messaged support and it would have worked great for my project!
